ui/components/app/add-network/add-network.js

```diff
@@ -250,7 +252,22 @@ const AddNetwork = () => {
                     type="inline"
                     className="add-network__add-button"
                     onClick={async () => {
-                      await dispatch(requestAddNetworkApproval(item, true));
+                      await dispatch(
+                        requestUserApproval({
+                          origin: ORIGIN_METAMASK,
+                          type: MESSAGE_TYPE.ADD_ETHEREUM_CHAIN,
+                          requestData: {
+                            chainId: item.chainId,
+                            rpcUrl: item.rpcUrl,
+                            ticker: item.ticker,
+                            rpcPrefs: item.rpcPrefs,
+                            imageUrl: item.rpcPrefs.imageUrl,
```

Comment:
Are we sure that `rpcPrefs` is defined here? Given that it's an optional property in our config.

I see that this block is meant for the default/popular networks and maybe they all have this defined, but that doesn't seem guaranteed from this code.

Reply:
Yeah we should probably protect against that here!

Reply:
fixed here: https://github.com/MetaMask/metamask-extension/pull/17421/commits/1b56e9d5d98ba1ff8bdcab67269672ed66c6a1cd

Elsewhere in the component we were accessing the `imageUrl` property on rpcPrefs without optional chaining. I've added it there too